chainer/functions/connection/linear.py

```diff
@@ -69,6 +69,81 @@ def backward(self, indexes, grad_outputs):
         return ret
 
 
+class LinearGradData(function_node.FunctionNode):
+
+    def __init__(self, linear):
+        return
+
+    def forward(self, inputs):
+        self.retain_inputs((0, 1))
+        gy, W = inputs
+
+        if not type_check.same_types(*inputs):
+            raise ValueError('numpy and cupy must not be used together\n'
+                             'type(gy): {0}, type(W): {1}'
+                             .format(type(gy), type(W)))
+        if (isinstance(gy, numpy.ndarray) and
+                not (gy.flags.c_contiguous or gy.flags.f_contiguous) and
+                1 in gy.shape):
+            gy = numpy.ascontiguousarray(gy)
+
+        gx = gy.dot(W).astype(gy.dtype, copy=False)
+        self.retain_inputs((0, 1))
+        return gx,
+
+    def backward(self, indexes, grad_outputs):
+        gy, W = self.get_retained_inputs()
+        ggx, = grad_outputs
+
+        ret = []
+        if 0 in indexes:
+            ggy = linear(ggx, W)
+            ret.append(chainer.functions.cast(ggy, gy.dtype))
+        if 1 in indexes:
+            gw, = LinearGradWeight(self).apply((ggx, gy))
+            ret.append(chainer.functions.cast(gw, W.dtype))
+        if 2 in indexes:
```

Comment:
Is this block by mistake? (there are only 2 inputs for LinearGradData)

Reply:
yes, gb is not need in  grad data double grad, I will remove it, thanks